adapter/slack.py

```diff
@@ -15,3 +15,24 @@ def __init__(self, app: App) -> None:
     # pragma: no cover
     async def handle_events(self, req: Request):
         return await self.handler.handle(req)
+
+    async def ask(self, request: Request):
+        data = await request.form()

```

Comment:
Looks good to me dul, can I ask since I don't quite understand yet - is this async/await is necessary? I'm guessing this is dependent on Slack's API?

Reply:
> Implementation of a Slack bot that is currently able to display questions asked by users.

Any particular step by steps to get the bot up and running on slack?

Reply:
Yes, this is necessary because request.form() returns an awaitable object, so the data won't be obtained if it's not awaited. I'm not entirely sure about this, but it has something to do with the interaction between FastAPI and the Slack API.

Reply:
OK sounds good to me, thanks for enlightening me. What about the running the slack bot?

Reply:
If you only want to add it to Slack, you can do this by opening the app from the Slack API website. Then, you can select the workspace to add the app. Currently, this app cannot be found directly in the list of Slack apps because it has not been released yet.

Reply:
I see, so we don't have to host it ourselves? I think I wrongly assumed it would be like a Discord bot that you have to run like any web server

Reply:
Oh, sorry, I thought you meant adding an existing bot. To set up a bot from scratch, you still need a web server to listen for events or process slash commands.

Reply:
Ok, thanks for clarifying